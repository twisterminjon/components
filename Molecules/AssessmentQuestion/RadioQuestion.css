.assessment-radio {
  color: var(--base__body_fg);
  cursor: pointer;
  padding: 20px 0 20px 0;
}

.assessment-radio input[type='radio'] {
  position: absolute;
  opacity: 0;
}

.assessment-radio input[type='radio'] + .assessment-radio-label:before {
  content: '';
  background: var(--assessment__input_bg);
  border-radius: 100%;
  border: solid 1px var(--assessment__input_border_color);
  display: inline-block;
  width: 22px;
  height: 22px;
  position: relative;
  top: -0.2em;
  margin-right: 1em;
  vertical-align: top;
  cursor: pointer;
  text-align: center;
  transition: all 250ms ease;
}

.assessment-radio input[type='radio']:checked + .assessment-radio-label:before {
  background-color: var(--brandcolor);
  box-shadow: inset 0 0 0 4px var(--assessment__input_bg);
}

.assessment-radio input[type='radio'] + .assessment-radio-label:empty:before {
  margin-right: 0;
}

.assessment-radio-label {
  font-weight: var(--font-base--regular);
  font-size: var(--font-base-body--size-medium);
  cursor: pointer;
}
