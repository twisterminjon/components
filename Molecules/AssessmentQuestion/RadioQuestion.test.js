import React from 'react'

import RadioQuestion from './RadioQuestion'
import { RadioQuestionData } from '../../../Mocks/Assessments.mock'

describe('RadioQuestion', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<RadioQuestion data={RadioQuestionData.options} onChange={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })
})
