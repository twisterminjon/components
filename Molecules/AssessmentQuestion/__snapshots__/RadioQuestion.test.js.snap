// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioQuestion matches the snapshot 1`] = `
<div>
  <RadioButton
    checked={false}
    data-testid="assessment-data-0"
    id={0}
    key="0"
    name="Suspendisse convallis dui ac dolor luctus, nec volutpat neque accumsan"
    onChangeHandler={[Function]}
  />
  <RadioButton
    checked={false}
    data-testid="assessment-data-1"
    id={1}
    key="1"
    name="Quisque nec lacinia purus, ut faucibus dolor"
    onChangeHandler={[Function]}
  />
  <RadioButton
    checked={false}
    data-testid="assessment-data-2"
    id={2}
    key="2"
    name="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
    onChangeHandler={[Function]}
  />
</div>
`;
